beta_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext_test
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/mathext"
+)
+
+func fequal(a, b float64) bool {
+	return (a - b) < 1e-12
+}
+
+func TestBeta(t *testing.T) {
+	{
+		v := mathext.Beta(1, 2)
+		u := mathext.Beta(2, 1)
+		if !fequal(v, u) {
+			t.Errorf("Beta(x,y) != Beta(y,x) (v=%v u=%v)\n", v, u)
+		}
+		const want = 0.5 // obtained from scipy.special.beta(1,2) (version=0.18.0)
+		if !fequal(v, want) {
+			t.Errorf("Beta(1,2) = %v. want=%v\n", v, want)
+		}
+	}
+
+	{
+		const (
+			p = 1
+			q = 2
+		)
+		u := mathext.Beta(p, q)
+		v := beta2(p, q)
+		if !fequal(u, v) {
+			t.Errorf("Beta(p,q) != Γ(p)Γ(q) / Γ(p+q) (v=%v u=%v)\n", v, u)
+		}
+	}
+	{
+		const (
+			p = 10
+			q = 20
+		)
+		v := mathext.Beta(p, q)
+		up1q := mathext.Beta(p+1, q)
+		upq1 := mathext.Beta(p, q+1)
+		u := up1q + upq1
+		if !fequal(v, u) {
+			t.Errorf("Beta(p,q) != Beta(p+1,q) + Beta(p,q+1) (v=%v u=%v)\n", v, u)
+		}
+	}
+
+	{
+		const (
+			p = +0
+			q = 10
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsInf(v, +1) {
+			t.Errorf("Beta(+0,10)=%v. want=+Inf\n", v)
+		}
+	}
+
+	{
+		const (
+			p = -0
+			q = 10
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsInf(v, +1) {
+			t.Errorf("Beta(-0,10)=%v. want=+Inf\n", v)
+		}
+	}
+
+	{
+		const (
+			p = +0
+			q = +0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(+0,+0)=%v. want=NaN\n", v)
+		}
+	}
+
+	{
+		var (
+			p = math.NaN()
+			q = 10.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(NaN,10)=%v. want=NaN\n", v)
+		}
+	}
+	{
+		var (
+			p = math.NaN()
+			q = +0.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(NaN,0)=%v. want=NaN\n", v)
+		}
+	}
+	{
+		var (
+			p = -1.0
+			q = +0.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(-1,0)=%v. want=NaN\n", v)
+		}
+	}
+	{
+		var (
+			p = -1.0
+			q = +1.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(-1,1)=%v. want=NaN\n", v)
+		}
+	}
+	{
+		var (
+			p = +0.0
+			q = -1.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(-1,0)=%v. want=NaN\n", v)
+		}
+	}
+	{
+		var (
+			p = +1.0
+			q = -1.0
+		)
+		v := mathext.Beta(p, q)
+		if !math.IsNaN(v) {
+			t.Errorf("Beta(-1,1)=%v. want=NaN\n", v)
+		}
+	}
+}
+
+func beta1(x, y float64) float64 {
+	return mathext.Beta(x, y)
+}
+
+func beta2(x, y float64) float64 {
+	return math.Gamma(x) * math.Gamma(y) / math.Gamma(x+y)
+}
+
+func betaBench(b *testing.B, f func(a, b float64) float64) {
```

Comment:
well... I could do:

```go
func BenchmarkBeta1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = mathext.Beta(10, 20)
	}
}

func BenchmarkBeta2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = math.Gamma(10) * math.Gamma(20) / math.Gamma(10+20)
	}
}
```

but the point of the benchmark is to assess the relative performance between `beta1` and `beta2`, so having 1 extra function call for the both of them is fine, I think.

don't you think? (I can easily be convinced otherwise, though)